AppServices/Validators/CreateCustomerRequestValidator.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Application.DTOs;
+using FluentValidation;
+using System;
+
+namespace Alan_WarrenDesafio1.Validators
+{
+    public class CreateCustomerRequestValidator : AbstractValidator<CreateCustomerRequest>
+    {
+        public CreateCustomerRequestValidator()
+        {
+            RuleFor(x => x.FullName)
+                .NotEmpty()
+                .WithMessage("Full name must not be null or empty")
+                .MinimumLength(2)
+                .WithMessage("Full name length must be more than 2 digits")
+                .MaximumLength(300)
+                .WithMessage("Full name length must cannot have more than 50 digits");
+
+            RuleFor(x => x.Email)
+                .NotEmpty()
+                .WithMessage("Email must not be null or empty")
+                .MinimumLength(9)
+                .WithMessage("Email length must be more than 9 digits")
+                .MaximumLength(256)
+                .WithMessage("Email length must cannot have more than 256 digits")
+                .Must(y => y.IsValidEmail())
+                .WithMessage("Email is invalid, please try again");
+
+            RuleFor(x => x)
+                .Must(x => x.Email == x.EmailConfirmation)
+                .WithMessage("Invalid email and email confirmation, please try again")
+                .Must(x => (x.EmailSms && !x.Whatsapp) || (!x.EmailSms && x.Whatsapp) || (x.EmailSms && x.Whatsapp))
+                .WithMessage("Tick at least one checkbox Email SMS or WhatsApp");
```

Comment:
Mensagem de erro incoerente com a validação
```suggestion
                .Must(x => (x.EmailSms && !x.Whatsapp) || (!x.EmailSms && x.Whatsapp) || (x.EmailSms && x.Whatsapp))
                .WithMessage("At least one or both 'EmailSms' or/and 'Whatsapp' must be true");
```